flink-runtime/pom.xml

```diff
@@ -179,6 +179,11 @@ under the License.
 			<artifactId>flink-test-utils-junit</artifactId>
 		</dependency>
 
+		<dependency>
+			<groupId>org.apache.flink</groupId>
+			<artifactId>flink-shaded-jackson-module-jsonSchema</artifactId>
+		</dependency>
```

Comment:
And if we move this dependency here, do we still need it in flink-runtime-web?

Reply:
1. The scope `test` cannot be used to transfer dependencies.
2. This dependency has been removed in flink-runtime-web.

Reply:
> The scope test cannot be used to transfer dependencies.

Then downstream modules (like flink-runtime-web / sql-gateway) should declare it again as a test dependency. That's just a downside of test-jars we have to live with.

Reply:
I see, so there are two options:
1. We add this dependency at `compile` scope here, so that downstream modules don't declare it again.
2. We add this dependency as `test` scope here, and the downstream modules need to declare it again to explicitly include the test-jar.

@WencongLiu, this PR currently goes for Option-1, while I'd in favor of Option-2 (and IIUC this is also what @zentol described) because it reduces the size of the production artifact should be more important.

Reply:
Done. The downstream modules have declared `flink-shaded-jackson-module-jsonSchema` as `test` scope in the latest fix-up. @xintongsong @zentol